src/Libraries/RevitNodes/Elements/Element.cs

```diff
@@ -193,8 +194,9 @@ protected ElementId InternalElementId
         public virtual void Dispose()
         {
 
-            // Do not cleanup Revit elements if we are shutting down Dynamo.
-            if (DisposeLogic.IsShuttingDown)
+            // Do not cleanup Revit elements if we are shutting down Dynamo or
+            // closing homeworkspace.
+            if (DisposeLogic.IsShuttingDown || DisposeLogic.IsClosingHomeworkspace)
```

Comment:
Although originally it is returning directly here, but I think it will be better if we unregister this element from the ElementIDLifecycleManager instance.